### PR TITLE
nix-search-cli: unstable-2023-09-12 -> v0.2.0+commit.7d6b4c5

### PR DESCRIPTION
**Pull request by peterldowns**
Upgrades `nix-search` to the latest version, with the following changes since 2023-09-12:

- `--help` has more useful information, including version info and a link to the github docs
- new `-r / --reverse` flag allows printing search results in reverse order, for making it easier to see relevant results

All changes are backwards compatible.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true` (unset, [defaults to true](https://nix.dev/manual/nix/2.18/command-ref/conf-file.html#conf-sandbox))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
  - `nix run nixpkgs#nixpkgs-review -- pr --post-result 37242` ~hangs indefinitely~ took forever and succeeded!
  -  From `rg`, there are no other packages in nixpkgs that depend on `nix-search-cli`.
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
  - `./result/bin/nix-search --help` shoes the version/commit message and the appropriate options.
  - `./result/bin/nix-search -p 'python3'` returned search results as expected
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
    - unnecessary, all changes are backwards-compatible
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).
  - I think so, this is my first time. I added myself to `maintainers`.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ni/nix-search-cli/package.nix [hunk 1]
```diff
@@ -1,33 +1,46 @@
...
-    rev = "f3f1c53c72dadac06472a7112aeb486ab5dda695";
-    hash = "sha256-YM1Lf7py79rU8aJE0PfQaMr5JWx5J1covUf1aCjRkc8=";
+    rev = "7d6b4c501ee448dc2e5c123aa4c6d9db44a6dd12";
+    hash = srcHash;
   };
+  inherit vendorHash;
 
-  vendorHash = "sha256-JDOu7YdX9ztMZt0EFAMz++gD7n+Mn1VOe5g6XwrgS5M=";
-
-  passthru.updateScript = unstableGitUpdater {
```
Comment (peterldowns) [inline pkgs/by-name/ni/nix-search-cli/package.nix:20]: I couldn't figure out how to use this or make it work, so I removed it.

File: pkgs/by-name/ni/nix-search-cli/package.nix [hunk 2]
```diff
@@ -1,33 +1,46 @@
...
 
-buildGoModule {
+let
+  # latest release version, without "v" prefix and "+commit" build metadata
+  # semver information:
+  # https://github.com/peterldowns/nix-search-cli/releases/tag/v0.2%2Bcommit.7d6b4c5
+  version = "0.2.0";
+  commit = "7d6b4c5";
+  srcHash = "sha256-0Zms/QVCUKxILLLJYsaodSW64DJrVr/yB13SnNL8+Wg=";
+  vendorHash = "sha256-RZuB0aRiMSccPhX30cGKBBEMCSvmC6r53dWaqDYbmyA=";
```
Comment (peterldowns) [inline pkgs/by-name/ni/nix-search-cli/package.nix:13]: to come up with these, i replaced them with `lib.fakeHash` and then ran 

```
NIX_PATH=nixpkgs=/Users/pd/code/nixpkgs nix-build -E 'with import <nixpkgs> {}; callPackage ./package.nix {}'
```
from inside the `pkgs/by-name/ni/nix-search-cli/` directory. Each time the build failed with a mismatched hash error, I replaced the `lib.fakeHash` with whatever hash it expected.

**Comment by peterldowns** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 347778`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>nix-search-cli</li>
  </ul>
</details>

**Comment by peterldowns** [comment]
@donovanglover thanks again for getting `nix-search` into nixpkgs in the first place. Hopefully this does the trick, but if you object to any of the changes, I'm happy to close this PR and let you update it instead.

**Comment by bloominstrong** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 347778`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>nix-search-cli</li>
  </ul>
</details>

**Comment by donovanglover** [comment]
@peterldowns Thank you for updating this! @azuwis has left a great review and would love to see those changes applied. Your PR should also have 2 commits in this order:

1. `maintainers: add peterldowns`
2. `nix-search-cli: 0-unstable-2023-09-12 -> 0.2.0`

**Comment by peterldowns** [comment]
@donovanglover I'm not going to bother — the update script stuff seems overly complicated and under documented. I couldn't get it to work; it's definitely over my head. I'm going to close this PR, it was "a learning experience" as the kids say.

**Comment by peterldowns** [comment]
I should add, thank you for making nix-search available for non-flakes users, and thanks for taking the time to review. I just don't have the wherewithal to get deeper into the Nix world — feels like pulling teeth every time I try. I sincerely appreciate your work and the work of all the other maintainers, being able to `nix profile install nixpkgs#whatever` is fantastic.

**Comment by donovanglover** [comment]
No worries. I actually [missed](https://github.com/NixOS/nixpkgs/pull/344603) an earlier PR which is now merged. Thank you!

**Pull request closed without merging**